bin/test/i18n/utils/test_sync_in_base.rb

```diff
@@ -10,6 +10,11 @@
       described_class.any_instance.expects(:process).once
       described_class.perform
     end
+
+    it 'calls report_runtime metrics with class name' do
+      I18n::Metrics.expects(:report_runtime).with(described_class.name, 'sync-in').once
+      described_class.perform
+    end
```

Comment:
It covers only one condition `... || name`, please test also this: `name[/^.*::(\w+::\w+)::SyncOut/, 1]`

Reply:
Done I belive!